### ISSUE TITLE
[2.0] Move providers to separate projects

### ISSUE THREAD

**Issue by stevenmaguire** [thephpleague/oauth1-client#43]
@bencorlett Now that the develop branch is fresh with some of the new architecture I wanted to bring up the topic of moving the provider implementations to independent repositories. I noticed that there are some empty repos within thephpleague organization. Were those created with this purpose in mind?
- [x] [oauth1-twitter](https://github.com/thephpleague/oauth1-twitter) maintained by @bencorlett 
- [x] [oauth1-trello](https://github.com/thephpleague/oauth1-trello) maintained by @stevenmaguire
-  [x] [oauth1-bitbucket](https://github.com/thephpleague/oauth1-bitbucket) maintained by @stevenmaguire
- [ ] oauth1-xing
- [ ] oauth1-uservoice
- [ ] oauth1-tumblr
- [ ] oauth1-magento

I am happy to assume responsibility for a few of them to get the ball rolling. I'd prefer to take trello, tumblr, twitter, and bitbucket, as I have experience with those APIs.

What are your thoughts?

**Comment by bencorlett** [comment]
Haha, those would be the only ones I’m interested in too. Yep, those repos were created to start adding those providers.

I think outside of those three, the others were community contributed :)

**Comment by stevenmaguire** [comment]
Well, I am in favor of two things. 1) We can split the four (trello, tumblr, twitter, and bitbucket) evenly and maintain under the league's banner 2) Delete the other empty repos and publish a new issue asking for maintains to take over.

If that sounds reasonable, I will amend my list of preferred providers to be trello and bitbucket :)

**Comment by bencorlett** [comment]
Sounds a plan!

**Comment by stevenmaguire** [comment]
Awesome! I assume you have rights to manage permissions on those projects. Could you add me as admin to oauth1-trello and oauth1-bitbucket? I will create issues for the remaining projects after I've got a boilerplate to "share".

**Comment by bencorlett** [comment]
Awesome! Doing that now :)

**Comment by stevenmaguire** [comment]
The `oauth1-trello` and `oauth1-bitbucket` packages are published and awaiting official release and/or further updates to this package. Once #50 is merged they will be fully functional as well.

If you are interested in picking up one of the incomplete packages listed in this issue, or another third-party, feel free to clone the trello or bitbucket repos for a boilerplate to get started.

**Comment by stevenmaguire** [comment]
I will take the remaining four providers (oauth1-xing, oauth1-uservoice, oauth1-tumblr, oauth1-magento) on once we get the develop branch somewhat stable and building successfully in CI.

**Comment by Zabanya** [comment]
can i help with magento?

**Comment by stevenmaguire** [comment]
Yes, please.
